src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -46,13 +46,27 @@ http_get_raw_response(
     return pHttpContext->GetResponse()->GetRawHttpResponse();
 }
 
-EXTERN_C __MIDL_DECLSPEC_DLLEXPORT VOID http_set_response_status_code(
+EXTERN_C __MIDL_DECLSPEC_DLLEXPORT HRESULT http_set_response_status_code(
     _In_ IHttpContext* pHttpContext,
     _In_ USHORT statusCode,
+    _In_ USHORT subStatusCode,
     _In_ PCSTR pszReason
 )
 {
-    pHttpContext->GetResponse()->SetStatus(statusCode, pszReason);
+    HRESULT hr S_OK;
+    IN_PROCESS_STORED_CONTEXT* pInProcessStoredContext = NULL;
+    hr = IN_PROCESS_STORED_CONTEXT::GetInProcessStoredContext(pHttpContext, &pInProcessStoredContext);
+    if (FAILED(hr))
+    {
+        return hr;
+    }
+
+    if (pInProcessStoredContext->QueryRequstAborted())
+    {
+        return E_ABORT;
+    }
+
+    return pHttpContext->GetResponse()->SetStatus(statusCode, pszReason, subStatusCode);
```

Comment:
What would happen if you called SetStatus after ResetConnection? How about WriteEntityChunks? Does it matter?
I wonder if we can skip all of this new API and call ResetConnection directly from managed.

Reply:
Calling SetStatus and WriteEntityChunks will succeed after calling ResetConnection. It is just being written to a local buffer. 
@pan-wang as part of the redesign of the pInvokes, will we be directly exposing HttpContext methods? If so, I'd agree it is better to have many of these calls in managed as currently most are wrappers for the native calls.